### DISCUSSION TITLE
Migrations issue

### DISCUSSION THREAD

**Discussion by guesswh0** [BeanieODM/beanie#405] [Question]
My project layout:
```shell
project_root:
    app/
        __init__.py
        models.py
    migrations/
    . . .
```

```python
# models.py
from beanie import Document, Indexed

class User(Document):
    username: Indexed(str, unique=True)
```

Running `beanie new-migration -n 1 -p migrations/` from project root creates migration file with 2 empty classes.

```python
# *_1.py
class Forward:
    ...

class Backward:
    ...
``` 
Initial migration file have to be empty?
Comparing to django migrations i fill like im missing something...

Lets consider `models.py` changed  to:
```python
# models.py
from beanie import Document, Indexed
from datetime import datetime

class User(Document):
    username: Indexed(str, unique=True)
    created: datetime = Field(default_factory=datetime.now)
```

What to do next?

**Comment by sblvkr** [comment]
I have the same problem. Help, please

**Comment by sblvkr** [comment]
```
def new_migration(name, path):
    path = Path(path)
    ts_string = datetime.now().strftime("%Y%m%d%H%M%S")
    file_name = f"{ts_string}_{name}.py"
    
    shutil.copy(template.__file__, path / file_name)
```
Do I understand correctly that there is no automatic creation of migrations? That is, the script simply creates an empty migration file, and I need to fill it out manually, based on the [examples](https://github.com/roman-right/beanie/tree/main/tests/migrations/migrations_for_test)?

**Reply by drcontai** [reply]
I'm new to Beanie and I was wondering about the same thing.  Did you confirm if that is the case?

**Reply by elmorem** [reply]
same!  this seems so weird to me.  it should either have some sortof startup script or at least much better examples.

**Reply by staticxterm** [reply]
Hi all, sorry for the inconvenience.
Beanie does not have any examples at the moment, apart from those that are in its "tests" part of the codebase.
You're supposed to write your own migration code.

**Comment by daniyal-preesoft** [comment]
I can only assume beanie doesn't really support this, as evident by the lack of response on this, despite the OP just following the documentation.
